### ISSUE TITLE
Research of existing implementations by language

### ISSUE THREAD

**Issue by Relequestual** [json-schema-org/community#418]
@Julian to define define success criteria, deliverables, and breakdown of tasks.

Assessed as high impact/low effort during our [collaborators summit 2023](https://github.com/orgs/json-schema-org/discussions/391).

**Comment by Julian** [comment]
(A brief task list is now in the issue description!)

**Comment by gregsdennis** [comment]
Have we decided on what "high quality implementation" means?

**Comment by Relequestual** [comment]
Could a work product of this include a published report? Target readership being technical leads and senior leadership.

**Comment by benjagm** [comment]
As peer the last OCWM #429:
@Julian it would be great if we add action items to this plan to describe what is our understanding of high quality implementations.

**Comment by Julian** [comment]
Hey! Can you guys fill me in on what was meant by

> describe what is our understanding of high quality implementations.

Are you saying something more elaborate than the task in the description? I.e.

> Extract a crude "health" metric of how good support for JSON Schema is in the language based on version availability, maturity of the implementation, activity of development in the implementation and compliance with the specification (time boxed to not be fully in depth)

**Comment by gregsdennis** [comment]
Yeah that's pretty much it.  The question came up in the last call, and scanning the issue, I didn't see anything, so I commented.  Then @benjagm posted the same question as a meeting update.

**Comment by Julian** [comment]
Got it, here are a few more specifics then, which I'll also add to the description:

* does it fail some very large number of tests
* does it support recent versions of JSON Schema
* does it have recent releases
* does it have recent commits
* does it have open security warnings
* does its maintainer consider it production-ready (or explicitly the reverse)
* how many open issues and pull requests does it have

I don't know that I believe in any objective summary metric for these (i.e. one single "yes it's good" or "no it's not") -- I don't believe such a metric exists in general for software, and JSON Schema implementations being no different in my opinion -- someone using them may however use these factoids to decide whether it's good or not I think. Perhaps watching for "improvement" on that list though is a good way to measure maturity.

**Comment by Relequestual** [comment]
I think it's worth noting in any results we do publish, that things like "recent releases or commets" are not always good indicators of anything on their own. If an implementation is 100% compliant and not javascript (ha), it's probably fine and doesn't need updates.

**Comment by Julian** [comment]
100% agree -- I think the same thing is true for "how many open issues does it have", some projects have lots of feature requests and some not, I think these are the exact sort of things which make a full summary metric basically impossible/useless.

**Comment by delixfe** [comment]
For the "health" metric, the [JSON-Schema-Test-Suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite) should play an important part. The criteria could be:

- JSON-Schema-Test-Suite runs in the release flow
- the tests are green for every supported schema version

**Comment by gregsdennis** [comment]
@delixfe that's definitely working into the equation, however we also have to recognize that an implementation may pass a majority of the tests but still have open, unaddressed issues, or it won't be updated when new tests are added, or other things.

**Comment by Relequestual** [comment]
After a recent chat, @Julian will be writing up a PR that includes a reporting template for this work.
